dev/com.ibm.ws.microprofile.reactive.messaging_fat/fat/src/com/ibm/ws/microprofile/reactive/messaging/fat/kafka/partitions/LivePartitionTestBean.java

```diff
@@ -0,0 +1,99 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.microprofile.reactive.messaging.fat.kafka.partitions;
+
+import static org.junit.Assert.assertTrue;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import org.eclipse.microprofile.reactive.messaging.Acknowledgment;
+import org.eclipse.microprofile.reactive.messaging.Acknowledgment.Strategy;
+import org.eclipse.microprofile.reactive.messaging.Incoming;
+import org.eclipse.microprofile.reactive.messaging.Message;
+
+/**
+ *
+ */
+@ApplicationScoped
+public class LivePartitionTestBean {
+
+    public static final String CHANNEL_IN = "live-partition-test-in";
+    public static final int WORK_TIME = 100;
+    public static final int FINAL_MESSAGE_NUMBER = 9999;
+
+    ArrayList<ReceivedMessage> messages = new ArrayList<>();
+    private final CountDownLatch paritionsFinished = new CountDownLatch(2);
+
+    @Incoming(CHANNEL_IN)
+    @Acknowledgment(Strategy.MANUAL)
+    public CompletionStage<Void> receive(Message<String> message) throws InterruptedException {
+
+        Thread.sleep(WORK_TIME);
+
+        ReceivedMessage status = new ReceivedMessage(message.getPayload());
+        messages.add(status);
+        System.out.println("Bean received message " + message.getPayload());
+
+        message.ack().handle((r, t) -> {
+            if (t == null) {
+                status.ackStatus.set(AckStatus.ACK_SUCCESS);
+                if (status.number == FINAL_MESSAGE_NUMBER) {
+                    paritionsFinished.countDown();
+                }
+            } else {
+                status.ackStatus.set(AckStatus.ACK_FAILED);
+            }
+            return null;
+        });
+
+        return CompletableFuture.completedFuture(null);
+    }
+
+    public void awaitFinish() throws InterruptedException {
+        assertTrue("Test bean did not process all messages within 30 seconds", paritionsFinished.await(30, TimeUnit.SECONDS));
```

Comment:
timeout constant ... based on the number of partitions??

Reply:
If anything it should be based on the number of messages.